Update migration and fix namespace inconsistencies

muhammadmirzo-pixel opened the pull request.

muhammadmirzo-pixel committed:
Update project to use PostgreSQL and enhance user model

- Replaced `Microsoft.EntityFrameworkCore.SqlServer` with `Npgsql.EntityFrameworkCore.PostgreSQL` in `DollarToSomConverter.csproj`.
- Added `FirstName` and `LastName` properties to the `User` class.
- Updated migration files to include the new `User` entity and its properties.
- Modified `AppDbContextModelSnapshot` to reflect the updated model structure.
- Configured PostgreSQL database context in `Program.cs`.
- Enhanced `AuthService` for user management operations.
- Created/modified `IConversionService` and `ConversionService` for conversion operations.
- Defined `IAuthService` interface for user management methods.
- Updated `appsettings.json` with a new PostgreSQL connection string.

muhammadmirzo-pixel committed:
Fixed conflicts and added stashed changes

muhammadmirzo-pixel committed:
Update migration and fix namespace inconsistencies

- Added new migration `SecondMigration`, replacing `InitialCreate`.
- Corrected namespace from `Service_folder` to `ServiceFolder` in multiple service files.
- Improved code organization and readability in `AuthService` and `IAuthService`.

Nozimjon04 commented on DollarToSomConverter/Service folder/IAuthServices/IAuthService.cs:
Nega IAuthServiceda Userlarni CRUD turibdi U IUserService da bolishi kerekku. IAuthServiceda authorizatsiya bilan bogliq ishlar qilinadi

Nozimjon04 commented on DollarToSomConverter/Service folder/AuthService/AuthService.cs:
Bu yerda ham UserServiceni ichida bolishi kerek edi

Nozimjon04 commented on DollarToSomConverter/Service folder/AuthService/AuthService.cs:
```diff
@@ -0,0 +1,75 @@
+public class AuthService : IAuthService
+{
+    private readonly IRepository<User> userRepository;
+    private readonly AppDbContext appDbContext;
```
Repositoryni ichida Savechangeasynchni yasab olib repositorydan olib kelib ishlating. AppDbContextni ham UserReponi olib kelishdan mano yoq sababi appDbContext.Users qilib bemalol ishlatorasiz u xolatda UserRepo kerek emas boladi. Shuning uchun IRepositoryni olib keling uning ichiga savechangeasyncni implement qlib oling

Nozimjon04 commented on DollarToSomConverter/Service folder/AuthService/AuthService.cs:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    public async Task<User> CreateAsync(User user)
+    {
```
User modelni ozidan foydalanmen bu yerda automatik toldiradigon polyalar bor masalan Id yoki CreatedAt shuning uchun UserForCreationDto madel yasab oling

Nozimjon04 commented on DollarToSomConverter/Service folder/AuthService/AuthService.cs:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    public async Task<User> DeleteAsync(long id)
+    {
```
nega delete qilganizdan keyin userni qaytaryapsiz?

Nozimjon04 commented on DollarToSomConverter/Service folder/AuthService/AuthService.cs:
```diff
@@ -0,0 +1,75 @@
+        return user; 
+    }
+
+    public async Task<User> UpdateAsync(User user, long id)
```
Updateda ham User uchun alohida model yasab oling. Userni ichida ozi Id bor yana alohida id yoz diyapsiz.!

Nozimjon04 commented on DollarToSomConverter/Service folder/IAuthServices/ConversionService.cs:
```diff
@@ -0,0 +1,47 @@
+    public ConversionService(IRepository<Conversion> repository, AppDbContext dbContext)
+    {
+        this.conversionRepository = repository;
+        this.dbContext = dbContext;
```
Bu yerda ham DbCotnext kerekmas

Nozimjon04 commented on DollarToSomConverter/Service folder/IAuthServices/ConversionService.cs:
```diff
@@ -0,0 +1,47 @@
+namespace DollarToSomConverter.Service_folder.IAuthServices;
+public class ConversionService : IConversionService
+{
+    private const decimal somConvert = 12850;
```
shu somConvertni static emas dynamic qilish kerek markazi bankdan olib kelib

Nozimjon04 commented on DollarToSomConverter/Service folder/IAuthServices/ConversionService.cs:
```diff
@@ -0,0 +1,47 @@
+    {
+        var conversion = await conversionRepository.GetByIdAsync(id);
+        return conversion ?? throw new Exception("Conversion not found");
+    }
```
**??** manosini bilasizmi?

muhammadmirzo-pixel committed:
Refactor repository and user management services

- Added `SaveChangeAsync` method to `IRepository` and implemented it in `Repository`.
- Updated `DollarToSomConverter.csproj` to disable nullable reference types.
- Modified `Auditable` class to ensure non-nullable `CreatedAt` and `UpdatedAt`.
- Removed all methods from `AuthService`, indicating a potential refactor.
- Redefined `IConversionService` with a new structure and namespace.
- Introduced `IUserService` interface for user management operations.
- Added new DTO classes for user data transfer: `UserForCreationDto`, `UserForResultDto`, `UserForUpdateDto`.
- Refactored `ConversionService` to utilize new repository methods.
- Implemented `UserService` to manage user-related operations with new DTOs.
- Added `Pagination` class to handle pagination parameters for user retrieval.
